### PR TITLE
test: Enhance ramps deeplink e2e

### PR DESCRIPTION
**Pull request by cortisiko**
## **Description**

This PR updates some actions in the ramps deep-linking e2e tests to reflect changes from an earlier PR that fixed a few edge case bugs. It also sets up an Android-only CI workflow for the ramps tests.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -699,6 +701,12 @@ workflows:
       - TEST_SUITE_TAG: '.*SmokeMultiChainPermissions.*'
     after_run:
       - android_e2e_test
+  run_tag_smoke_ramps_ios:
```
Comment (cortisiko) [inline bitrise.yml:704]: smoke workflow for ramps e2e. iOS only.

File: e2e/pages/Network/NetworkAddedBottomSheet.js [hunk 2]
```diff
@@ -13,23 +13,15 @@ class NetworkAddedBottomSheet {
...
 
   get switchNetworkButton() {
-    return device.getPlatform() === 'android'
-      ? Matchers.getElementByLabel(
-        NetworkAddedBottomSheetSelectorsIDs.SWITCH_NETWORK_BUTTON,
-        )
-      : Matchers.getElementByID(
-        NetworkAddedBottomSheetSelectorsIDs.SWITCH_NETWORK_BUTTON,
-        );
+    return Matchers.getElementByID(
```
Comment (cortisiko) [inline e2e/pages/Network/NetworkAddedBottomSheet.js:16]: updating selectors for both iOS and android. Locating byID across both platforms

File: e2e/resources/networks.e2e.js [hunk 3]
```diff
@@ -46,7 +46,7 @@ const PopularNetworksList = {
       type: 'rpc',
       chainId: toHex('10'),
       rpcUrl: `https://optimism-mainnet.infura.io/v3/${infuraProjectId}`,
-      nickname: 'Optimism',
+      nickname: 'OP Mainnet',
```
Comment (cortisiko) [inline e2e/resources/networks.e2e.js:49]: updating network name

File: e2e/specs/ramps/deeplink-to-buy-flow-with-unsupported-network.spec.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ import TestHelpers from '../../helpers';
 
 import { loginToApp } from '../../viewHelper';
 import { withFixtures } from '../../fixtures/fixture-helper';
-import { SmokeCore } from '../../tags';
+import { SmokeRamps } from '../../tags';
```
Comment (cortisiko) [inline e2e/specs/ramps/deeplink-to-buy-flow-with-unsupported-network.spec.js:6]: tagging tests Ramps

File: e2e/tags.js [hunk 5]
```diff
@@ -8,6 +8,7 @@ const tags = {
   smokeAssets: 'smokeAssets:',
   smokeIdentity: 'smokeIdentity:',
   smokeMultiChainPermissions: 'SmokeMultiChainPermissions:',
+  smokeRamps: 'SmokeRamps:',
```
Comment (cortisiko) [inline e2e/tags.js:11]: ramps tag

File: e2e/resources/networks.e2e.js (hunk 3 above)
Comment (sethkfman) [reply e2e/resources/networks.e2e.js:49]: Can we get this list from application directory `metamask-mobile/app/util/networks/customNetworks.tsx`?

Comment (sethkfman) [reply e2e/resources/networks.e2e.js:49]: This is not a blocker to merge but a way to reduce effort in the future.

**Review by sethkfman — APPROVED** [review]
LGTM

File: e2e/resources/networks.e2e.js (hunk 3 above)
Comment (cortisiko) [reply e2e/resources/networks.e2e.js:49]: For sure, Here is a [ticket](https://github.com/MetaMask/metamask-mobile/issues/12825) to get that done.

**Pull request merged**